lib/runtime/errors/invalid_transaction.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package errors
+
+import (
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// InvalidTransaction is child VDT of TransactionValidityError
+type InvalidTransaction scale.VaryingDataType
+
+// Index fulfils the VaryingDataTypeValue interface.  T
+func (InvalidTransaction) Index() uint {
+	return 0
+}
+
+// Call The call of the transaction is not expected
+type Call struct{}
+
+// Index Returns VDT index
+func (Call) Index() uint { return 0 }
```

Comment:
Could you add a `String() string` or `Error() string` (up to you really) method here for each of these types?

That way you can just have

```go
func (i *InvalidTransaction) Error() string {
    value := i.Value()
    err, ok := value.(error)
    if !ok {
        panic(fmt.Sprintf("%T does not implement the error type", value))
    }
    return err.Error()
}
```